### PR TITLE
[VIRTS-2100] Fire events when operation changes state

### PR DESCRIPTION
**Pull request by bworrell**
## Description
Update `Operation` to fire an event whenever its state is changed.

Notes:
* Does not fire an event when setting the initial state
* Does not fire an event if setting the state to its current state

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
* All tests pass
* I added tests to ensure the event service was being invoked as expected

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tests/services/test_planning_svc.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def __init__(self, **kwargs):
 
 
 @pytest.fixture
-def setup_planning_test(loop, ability, agent, operation, data_svc, init_base_world):
+def setup_planning_test(loop, ability, agent, operation, data_svc, event_svc, init_base_world):
```
Comment (bworrell) [inline tests/services/test_planning_svc.py:60]: After making the operation state setter fire an event, the planning service tests started failing due to the event service not being instantiated/registered. I added this fixture to initialize the event service and make it discoverable.

File: app/objects/c_operation.py [hunk 2]
```diff
@@ -42,6 +46,9 @@ def build_planner(self, data, **_):
 
 
 class Operation(FirstClassObjectInterface, BaseObject):
+    EVENT_EXCHANGE = "operation"
+    EVENT_QUEUE_STATE_CHANGED = "state_changed"
+    EVENT_QUEUE_COMPLETED = "completed"
```
Comment (bworrell) [inline app/objects/c_operation.py:51]: Fixme: double quotes

File: app/objects/c_operation.py [hunk 3]
```diff
@@ -58,6 +65,27 @@ def states(self):
                     FINISHED='finished',
                     CLEANUP='cleanup')
 
+    @property
+    def state(self):
+        return self._state
+
+    @state.setter
+    def state(self, value):
+        previous_state = getattr(self, "_state", NO_PREVIOUS_STATE)
```
Comment (bworrell) [inline app/objects/c_operation.py:74]: Fixme: double quotes

**Review by b17zr — APPROVED** [review]
Excellent use of generators.

**Review by clenk — APPROVED** [review]
LGTM!

**Pull request merged**